pkg/controller/v1alpha2/applicationconfiguration/render.go

```diff
@@ -261,10 +271,18 @@ func SetWorkloadInstanceName(traitDefs []v1alpha2.TraitDefinition, w *unstructur
 		if c.Status.LatestRevision == nil {
 			return fmt.Errorf(errFmtCompRevision, c.Name)
 		}
-		// if revisionEnabled, use revisionName as the workload name
-		if err := pv.SetString(instanceNamePath, c.Status.LatestRevision.Name); err != nil {
+
+		componentLastRevision := c.Status.LatestRevision.Name
+		if currentWorkload.GetLabels()[oam.LabelAppComponentRevision] == componentLastRevision {
```

Comment:
currentWorkload is only needed if we switch from a named workload to a revisionEnable name so the main logic probably need to reflect that.

Reply:
Per the issue #282 "if we switch from a named workload to a revisionEnable name" without comparing to the current existing workload, the other workload, which doesn't have any changes, will be recreate. This should not happen.